### PR TITLE
gRPC services need to tolerate EAR packaging 

### LINKED ISSUE #18492: gRPC service registration broken for EAR deployments
**Describe the bug**  
gRPC services deployed on Liberty via `grpc-1.0` are not registered correctly when they are embedded in an EAR (rather than a WAR). This results in the services never being made available. Additionally, the following FFDC is logged:

```
Exception = com.ibm.wsspi.adaptable.module.UnableToAdaptException
Source = io.openliberty.grpc.internal.servlet.GrpcServerComponent
probeid = 230
Stack Dump = com.ibm.wsspi.adaptable.module.UnableToAdaptException: CWWKM0453E: WebSphere Application Server internal error occurred. Please contact WebSphere Application Server support with the following data: Container is not a module com.ibm.ws.adaptable.module.internal.InterpretedContainerImpl@1e2a2e4f ( <app_location_and_name> )
	at com.ibm.ws.container.service.annocache.internal.WebAnnotationsAdapterImpl.adapt(WebAnnotationsAdapterImpl.java:54)
	at com.ibm.ws.container.service.annocache.internal.WebAnnotationsAdapterImpl.adapt(WebAnnotationsAdapterImpl.java:33)
	at com.ibm.ws.adaptable.module.internal.AdapterFactoryServiceImpl.adapt(AdapterFactoryServiceImpl.java:200)
	at com.ibm.ws.adaptable.module.internal.AdaptableContainerImpl.adapt(AdaptableContainerImpl.java:174)
	at com.ibm.ws.adaptable.module.internal.InterpretedContainerImpl.adapt(InterpretedContainerImpl.java:203)
	at com.ibm.ws.container.service.annocache.AnnotationsBetaHelper.getWebAnnotations(AnnotationsBetaHelper.java:268)
	at io.openliberty.grpc.internal.servlet.GrpcServerComponent.initServicesHelper(GrpcServerComponent.java:243)
        ...
```

**Steps to Reproduce**  
Deploy an EAR archive with bundled gRPC services

**Expected behavior**  
The gRPC services should start as expected and no FFDCs should be logged

**Diagnostic information:**  
 - OpenLiberty Version: All
 - Affected feature(s) `grpc-1.0`
 - Java Version: N/A

### PR DESCRIPTION
**Pull request by wtlucy**
Fixes #18492
- update app startup annotation scanning code in `GrpcServerComponent` to work with EAR applications
- fix whitespace issues

### CODE REVIEW

**Comment by wtlucy** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.grpc_fat

**Comment by wtlucy** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.grpc_fat

**Comment by wtlucy** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.grpc_fat

**Comment by wtlucy** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.grpc_fat

File: dev/com.ibm.ws.grpc_fat/test-applications/HelloWorldApp.ear/resources/META-INF/application.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
```
Comment (loriadi) [inline dev/com.ibm.ws.grpc_fat/test-applications/HelloWorldApp.ear/resources/META-INF/application.xml:5]: copyright date?

**Review by loriadi — APPROVED** [review]

**Comment by wtlucy** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.grpc_fat

File: dev/com.ibm.ws.grpc_fat/test-applications/HelloWorldApp.ear/resources/META-INF/application.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                <context-root>HelloWorldService</context-root>
+            </web>
+        </module>
+        <module id="WebModule_2">
+            <web>
+                <web-uri>HelloWorldClient.war</web-uri>
+                <context-root>HelloWorldClient</context-root>
+            </web>
+        </module>
+</application>
```
Comment (volosied) [inline dev/com.ibm.ws.grpc_fat/test-applications/HelloWorldApp.ear/resources/META-INF/application.xml]: New Line

File: dev/io.openliberty.grpc.1.0.internal/src/io/openliberty/grpc/internal/config/GrpcServiceConfigImpl.java [hunk 3]
```diff
@@ -38,175 +38,175 @@
...
+    }
+
+    public static void addApplication(String name) {
+        applications.add(name);
+    }
+
+    public static void removeApplication(String name) {
+        applications.remove(name);
+    }
 }
```
Comment (volosied) [inline dev/io.openliberty.grpc.1.0.internal/src/io/openliberty/grpc/internal/config/GrpcServiceConfigImpl.java]: New Line

**Review by volosied — APPROVED** [review]
Thanks for the changes. I think it looks pretty good overall.

File: dev/io.openliberty.grpc.1.0.internal/src/io/openliberty/grpc/internal/servlet/LibertyAuthorizationInterceptor.java [hunk 4]
```diff
@@ -25,21 +25,21 @@
...
+        String key = headers.get(GrpcServerSecurity.LIBERTY_AUTH_KEY);
+        if (!GrpcServerSecurity.isAuthorized(key)) {
+            call.close(Status.UNAUTHENTICATED.withDescription("Unauthorized"), headers);
+            // return no-op listener
+            return new ServerCall.Listener<ReqT>() {
+            };
+        }
+        return next.startCall(call, headers);
+    }
 }
```
Comment (volosied) [inline dev/io.openliberty.grpc.1.0.internal/src/io/openliberty/grpc/internal/servlet/LibertyAuthorizationInterceptor.java]: Here too

**Comment by wtlucy** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.grpc_fat

**Comment by wtlucy** [comment]
#run-libby-bot

**Pull request merged**